src/modules/siptrace/siptrace.c

```diff
@@ -1686,10 +1692,14 @@ static void trace_onreply_out(struct cell *t, int type, struct tmcb_params *ps)
 
 	if (info->uriState == STRACE_RAW_URI) {
 		LM_BUG("uriState must be either UNUSED or PARSED here! must be a bug! Message won't be traced!\n");
-		return;
+		goto end;
 	}
 
 	sip_trace_store(&sto, info->uriState == STRACE_PARSED_URI ? &info->u.dest_info : NULL, NULL);
+
+end:
+	if (faked && parsed_f) 
+		free_from(msg->from->parsed);
```

Comment:
At least for safety,  probably `msg->from->parsed` should be set back to NULL, not to be freed again in other places.